include/oneapi/dpl/internal/random_impl/uniform_real_distribution.h

```diff
@@ -226,9 +226,8 @@ class uniform_real_distribution
     generate(_Engine& __engine, const param_type& __params)
     {
         auto __engine_output = __engine();
-        auto __res = static_cast<scalar_type>(__engine_output);
-        __res = ((__res - static_cast<scalar_type>(__engine.min())) /
-                 (1 + static_cast<scalar_type>(__engine.max() - __engine.min()))) *
+        scalar_type __res = ((__engine_output - __engine.min()) /
+                 static_cast<scalar_type>(1 + (__engine.max() - __engine.min()))) *
```

Comment:
Here and at the next places the question about overflow for "1 + (__engine.max() - __engine.min())"